### PR TITLE
Fix/anchor

### PR DESCRIPTION
**Pull request by minorOffense**
Add more functional adjustments to match the pathauto / transliteration settings in Drupal.

See https://github.com/CityofOttawa/ottawa_profile/issues/1040

### CODE REVIEW

**Comment by minorOffense** [comment]
Luc put his settings on ottv3. But I made the changes. Just so we know who fixed what :-P

**Comment by minorOffense** [comment]
Also changed it so it loads the minified version of the js. I minified it before committing.

https://github.com/CityofOttawa/ottawa_anchor_menu/commit/6583bdcddcc0ed0fa6a4cb0a67a9c60a2044583f

File: dist/bootstrap-toc.js [hunk 1]
```diff
@@ -48,8 +49,10 @@
         anchor = anchor.replace(/[œ]/g, 'oe');
         anchor = anchor.replace(/[']/g, '');
         anchor = anchor.replace(/[’]/g, '');
-        anchor = anchor.replace(/[^A-Za-z0-9]+/g, '-');
+        anchor = anchor.replace(/\ba\b|\ban\b|\bas\b|\bat\b|\bbefore\b|\bbut\b|\bby\b|\bfor\b|\bfrom\b|\bis\b|\bin\b|\binto\b|\blike\b|\bof\b|\boff\b|\bon\b|\bonto\b|\bper\b|\bsince\b|\bthan\b|\bthe\b|\bthis\b|\bthat\b|\bto\b|\bup\b|\bvia\b|\bwith\b/g, '');
+        anchor = anchor.replace(/\s+/g,'-');
         anchor = anchor.replace(/([~!@#$%^&*()_+=`{}\[\]\|\\:;'<>,.\/? ])+/g, '-').replace(/^(-)+|(-)+$/g,'');
+        anchor = anchor.replace(/[^A-Za-z0-9]+/g, '-');
```
Comment (minorOffense) [inline dist/bootstrap-toc.js:55]: Had to change the order of the replacements so we didn't end up with '--' in the resulting string.

File: dist/bootstrap-toc.js [hunk 2]
```diff
@@ -48,8 +49,10 @@
         anchor = anchor.replace(/[œ]/g, 'oe');
         anchor = anchor.replace(/[']/g, '');
         anchor = anchor.replace(/[’]/g, '');
-        anchor = anchor.replace(/[^A-Za-z0-9]+/g, '-');
+        anchor = anchor.replace(/\ba\b|\ban\b|\bas\b|\bat\b|\bbefore\b|\bbut\b|\bby\b|\bfor\b|\bfrom\b|\bis\b|\bin\b|\binto\b|\blike\b|\bof\b|\boff\b|\bon\b|\bonto\b|\bper\b|\bsince\b|\bthan\b|\bthe\b|\bthis\b|\bthat\b|\bto\b|\bup\b|\bvia\b|\bwith\b/g, '');
+        anchor = anchor.replace(/\s+/g,'-');
```
Comment (minorOffense) [inline dist/bootstrap-toc.js:53]: This replaces all sequential spaces with a single '-'

File: dist/bootstrap-toc.js [hunk 3]
```diff
@@ -48,8 +49,10 @@
         anchor = anchor.replace(/[œ]/g, 'oe');
         anchor = anchor.replace(/[']/g, '');
         anchor = anchor.replace(/[’]/g, '');
-        anchor = anchor.replace(/[^A-Za-z0-9]+/g, '-');
+        anchor = anchor.replace(/\ba\b|\ban\b|\bas\b|\bat\b|\bbefore\b|\bbut\b|\bby\b|\bfor\b|\bfrom\b|\bis\b|\bin\b|\binto\b|\blike\b|\bof\b|\boff\b|\bon\b|\bonto\b|\bper\b|\bsince\b|\bthan\b|\bthe\b|\bthis\b|\bthat\b|\bto\b|\bup\b|\bvia\b|\bwith\b/g, '');
```
Comment (minorOffense) [inline dist/bootstrap-toc.js:52]: Pathauto dynamically generates a regex to strip certain words from the path. This is a copy of the current settings. If the settings in pathauto change we have to change this.

We could consider sending the regex string as a js var that this file could read instead.

See pathauto.inc around like 200 or so.

**Review by walkerjam — APPROVED** [review]
Agree about sending a regex string (or array) as a var to keep current with pathauto, would be nice.

**Comment by minorOffense** [comment]
Ok for now we'll leave it. But I'll log an enhancement for it.

**Pull request merged**